### PR TITLE
Empty stim trace

### PR DESCRIPTION
**Pull request by sgratiy**
The first 4 commits is a refactorying: 
Splitting some modules in order to reduce size and make code more modular.

The last 2 commits are to fix a bug:
For sweeps truncated by the MIES  before stimulus start, the sweep constructor was getting an error. It was happening because sweep creation depended on the proper return from the get_exprimental epoch() in order to align time.  Solution: more aligning the time out of the sweep constructor into a method which is called at a later feature extraction stage.

### CODE REVIEW

File: ipfx/feature_record.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    base = cell_features["long_squares"]["rheobase_sweep"]["spikes"][0]
+    ephys_features["upstroke_downstroke_ratio_long_square"] = nan_get(base, "upstroke_downstroke_ratio")
+    ephys_features["peak_v_long_square"] = nan_get(base, "peak_v")
+    ephys_features["peak_t_long_square"] = nan_get(base, "peak_t")
+    ephys_features["trough_v_long_square"] = nan_get(base, "trough_v")
+    ephys_features["trough_t_long_square"] = nan_get(base, "trough_t")
+    ephys_features["fast_trough_v_long_square"] = nan_get(base, "fast_trough_v")
+    ephys_features["fast_trough_t_long_square"] = nan_get(base, "fast_trough_t")
+    ephys_features["slow_trough_v_long_square"] = nan_get(base, "slow_trough_v")
+    ephys_features["slow_trough_t_long_square"] = nan_get(base, "slow_trough_t")
```
Comment (njmei) [inline ipfx/feature_record.py]: Seems a bit repetitive maybe instead you could do something similar to:
```
features = ["upstroke_downstroke_ratio", "peak_v", ...]

postfix = '_long_square'
for ft in features:
    new_ft_name = '{}{}'.format(ft, postfix)
    ephys_features[new_ft_name] = nan_get(base, ft)
```

**Comment by njmei** [comment]
Maybe might be worthwhile to write a test or two to make sure no future regressions occur?

File: ipfx/feature_record.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+import numpy as np
+
+
+def build_cell_feature_record(cell_features, sweep_features):
```
Comment (nikahk) [inline ipfx/feature_record.py]: This function is doing multiple things. You should split the work.

File: ipfx/feature_record.py (hunk 1 above)
Comment (sgratiy) [reply ipfx/feature_record.py]: @njmei Good suggestion. Done.

File: ipfx/data_set_features.py [hunk 3]
```diff
@@ -123,18 +124,19 @@ def extract_sweep_features(data_set, sweep_table):
         sweep_numbers = sorted(sweep_numbers)
         logging.debug("%s:%s" % (stimulus_name, ','.join(map(str, sweep_numbers))))
 
-        sweep_set = data_set.sweep_set(sweep_numbers)
+#        sweep_set = data_set.sweep_set(sweep_numbers)
```
[the remark this answers is not in the document]
Comment (sgratiy) [reply ipfx/data_set_features.py]: @nikahk Done

File: ipfx/feature_record.py (hunk 2 above)
Comment (sgratiy) [reply ipfx/feature_record.py]: I refactored a bit to reduce repetition.

**Comment by sgratiy** [comment]
@t-b I agree, I need to add more tests. I will add them on the next PR (i promise :)). I would like to merge this PR asap though to enable processing cells which were previously failing.

File: ipfx/feature_record.py (hunk 2 above)
Comment (sgratiy) [reply ipfx/feature_record.py]: @nikahk Thanx for pointing out the "multiple thing" thing. I have refactored to stay close to single responsibility.

**Pull request merged**